Storage/Bookmarks/BookmarksModel.swift

```diff
@@ -62,6 +62,11 @@ public class BookmarkFolder: BookmarkNode {
     override public var canDelete: Bool {
         return false
     }
+
+    public func removeItemWithGUID(guid: GUID) -> MemoryBookmarkFolder {
```

Comment:
There are two places we might be using a 'folder'.

One is as part of a model, in its role as providing a title for the UI and a list of children.

The other is as a row in the display list, where it's simply a `BookmarkNode` subtype.

The factory method you spotted, and thus that `BookmarkFolder`, serves the latter purpose.

AFAICS both `itemFactory` and `folderFactory` could safely be changed to return `BookmarkNode`: just a thing with an icon and a title.

Using `SQLiteBookmarkFolder` wouldn't work — that needs a cursor of children, and is intended to play the former role.